components/ws-manager/pkg/manager/monitor.go

```diff
@@ -495,9 +485,16 @@ func actOnPodEvent(ctx context.Context, m actingManager, manager *Manager, statu
 			}
 		}
 
-		_, startedDisposal := wso.Pod.Annotations[startedDisposalAnnotation]
-		span.LogKV("startedDisposal", startedDisposal)
-		if terminated && !startedDisposal {
+		ds := &workspaceDisposalStatus{}
+		if rawDisposalStatus, ok := pod.Annotations[disposalStatusAnnotation]; ok {
+			err := json.Unmarshal([]byte(rawDisposalStatus), ds)
+			if err != nil {
+				log.WithError(err).Errorf("cannnot parse disposalStatusAnnotation: %s", rawDisposalStatus)
+			}
+		}
+
+		span.LogKV("disposalStatusAnnotation", ds)
+		if terminated && !ds.Status.IsDisposing() {
```

Comment:
What if we change change the function as
```go
func (ds DisposalStatus) IsDisposed() bool {
	return ds == DisposalFinished
}
```

and change this line as
```go
if terminated && !ds.Status.IsDisposed() {
```

so we could remove the line 518 `if ds.Status != DisposalFinished {`

The code be simpler, WDYT?

Reply:
@jenting Thanks for your comments. https://github.com/gitpod-io/gitpod/pull/12249